### DISCUSSION TITLE
Recent Company VPN Change broke extension!!

### DISCUSSION THREAD

**Discussion by ameeno** [mbnuqw/sidebery#1561] [General]
## Steps to reproduce
- Open Firefox in [new profile](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles)
- Install Sidebery
- setup container/(work personal regex matches.
- breaks vpn connection

## Expected behavior

VPN should work.
## Actual behavior
COntainer jumps and session login fails

## Debug info

<details><summary><b>Addon data</b></summary><code>

{
  "settings": {
    "version": "4.10.2",
    "nativeScrollbars": false,
    "selWinScreenshots": true,
    "tabsCheck": true,
    "tabsFix": "reinit",
    "markWindow": true,
    "markWindowPreface": "​",
    "ctxMenuNative": false,
    "autoHideCtxMenu": 1000,
    "ctxMenuRenderInact": true,
    "ctxMenuIgnoreContainers": "",
    "navBarLayout": "horizontal",
    "navBarInline": true,
    "hideAddBtn": true,
    "hideSettingsBtn": true,
    "navBtnCount": true,
    "hideEmptyPanels": true,
    "navActTabsPanelLeftClickAction": "none",
    "navActBookmarksPanelLeftClickAction": "scroll",
    "navMidClickAction": "none",
    "navSwitchPanelsWheel": true,
    "groupLayout": "list",
    "skipEmptyPanels": false,
    "dndTabAct": true,
    "dndTabActDelay": 50,
    "dndTabActMod": "none",
    "dndExp": "pointer",
    "dndExpDelay": 50,
    "dndExpMod": "none",
    "stateStorage": "global",
    "warnOnMultiTabClose": "any",
    "activateOnMouseUp": false,
    "activateLastTabOnPanelSwitching": true,
    "showTabRmBtn": true,
    "showTabCtx": true,
    "hideInact": false,
    "activateAfterClosing": "prev",
    "activateAfterClosingPrevRule": "tree",
    "activateAfterClosingNextRule": "tree",
    "activateAfterClosingGlobal": true,
    "activateAfterClosingNoFolded": false,
    "activateAfterClosingNoDiscarded": false,
    "shiftSelAct": true,
    "askNewBookmarkPlace": true,
    "tabsRmUndoNote": true,
    "nativeHighlight": true,
    "tabsUnreadMark": true,
    "tabsReloadLimit": 5,
    "tabsReloadLimitNotif": true,
    "tabsPanelSwitchActMove": true,
    "moveNewTabPin": "end",
    "moveNewTabParent": "default",
    "moveNewTabParentActPanel": true,
    "moveNewTab": "end",
    "pinnedTabsPosition": "panel",
    "pinnedTabsList": false,
    "pinnedAutoGroup": true,
    "tabsTree": true,
    "groupOnOpen": true,
    "tabsTreeLimit": "none",
    "hideFoldedTabs": false,
    "autoFoldTabs": true,
    "autoFoldTabsExcept": 5,
    "autoExpandTabs": true,
    "rmChildTabs": "none",
    "tabsChildCount": true,
    "tabsLvlDots": true,
    "discardFolded": true,
    "discardFoldedDelay": 15,
    "discardFoldedDelayUnit": "min",
    "tabsTreeBookmarks": true,
    "treeRmOutdent": "first_child",
    "bookmarksPanel": true,
    "warnOnMultiBookmarkDelete": "collapsed",
    "openBookmarkNewTab": true,
    "midClickBookmark": "open_new_tab",
    "actMidClickTab": true,
    "autoCloseBookmarks": true,
    "autoRemoveOther": false,
    "highlightOpenBookmarks": true,
    "activateOpenBookmarkTab": true,
    "showBookmarkLen": true,
    "bookmarksRmUndoNote": true,
    "fontSize": "s",
    "bgNoise": false,
    "animations": true,
    "animationSpeed": "fast",
    "theme": "default",
    "style": "dark",
    "sidebarCSS": "#root {  \n  --tabs-font: 1.1rem sans-serif;\n  --tabs-count-font: .7rem sans-serif;\n  --bookmarks-bookmark-font: .975rem sans-serif;\n  --bookmarks-folder-font: 1.1rem sans-serif;\n  --nav-btn-width: 42px;\n  --nav-btn-height: 42px;\n}\n\n/* Adjust styles according to sidebar width */\n@media screen and (max-width: 47px) {\n  #root {\n    --tabs-indent: unset;\n  }\n\n  .Tab[data-audible]:not([data-muted]) svg.-loud,\n  .Tab[data-muted] svg.-mute {\n    transform: translateY(4px) translateX(-13px);\n  }\n\n  .NavigationBar .panel-btn:not([data-active=\"true\"]),\n  .Sidebar .settings-btn,\n  .Tab .close,\n  .Tab .title {\n    visibility: collapse;\n  }\n\n  .Tab .exp,\n  .Tab[data-parent][data-folded] .fav {\n    pointer-events: none;\n  }\n\n}\n\n/*\n * Add margins and rounding around tabs\n */\n\n#root {\n  --tabs-height: 42px;\n}\n\n/* Background layer */\n.Tab .lvl-wrapper:after {\n  content: '';\n  position: absolute;\n  top: 5px;\n  left: 5px;\n  width: calc(100% - 10px);\n  height: calc(100% - 6px);\n  border-radius: 4px;\n  z-index: -1;\n}\n\n@media (prefers-color-scheme:light) {\n  #root {\n    --tabs-activated-bg: white !important;\n    --tabs-bg-active: var(--tabs-activated-bg) !important;\n    --tabs-selected-fg: var(--tabs-activated-fg) !important;\n    --tabs-selected-bg: var(--tabs-activated-bg) !important;\n    --bg: #f9f9fb !important;\n    --chrome-content-separator-color: rgb(204, 204, 204);\n  }\n\n  .Tab[data-selected] .lvl-wrapper:after,\n  .Tab[data-active] .lvl-wrapper:after {\n    box-shadow: 0 0 1px rgba(128, 128, 142, 0.9), 0 0 4px rgba(128, 128, 142, 0.5);\n  }\n}\n\n@media (prefers-color-scheme:dark) {\n  #root {\n    --tabs-activated-bg: #42414d !important;\n    --tabs-bg-active: var(--tabs-activated-bg) !important;\n    --tabs-selected-fg: var(--tabs-activated-fg) !important;\n    --tabs-selected-bg: var(--tabs-activated-bg) !important;\n    --bg: #2b2a33 !important;\n    --chrome-content-separator-color: #0c0c0d;\n    --tabs-bg-hover: rgb(51, 51, 62) !important;\n  }\n\n  .Tab[data-selected] .lvl-wrapper:after,\n  .Tab[data-active] .lvl-wrapper:after {\n    box-shadow: 0 0 1px rgba(128, 128, 142, 0.9);\n  }\n\n  .Tab .placeholder>svg {\n    fill: var(--nav-btn-fg);\n  }\n}\n\n/* Reset default styles */\n.Tab:hover,\n.Tab:active,\n.Tab[data-active],\n.Tab[data-active]:active,\n.Tab[data-selected],\n.Tab[data-selected]:hover,\n.Tab[data-selected]:active {\n  background: transparent;\n}\n\n/* Reapply styles */\n\n.Tab:hover .lvl-wrapper:after {\n  background-color: var(--tabs-bg-hover);\n}\n\n.NavigationBar .panel-btn[data-active=\"true\"],\n.Tab:active .lvl-wrapper:after,\n.Tab[data-active]:active .lvl-wrapper:after {\n  background-color: var(--tabs-bg-active);\n}\n\n.Tab[data-active] .lvl-wrapper:after {\n  background-color: var(--tabs-activated-bg);\n}\n\n\n.Tab[data-selected] .lvl-wrapper:after {\n  background-color: var(--tabs-selected-bg);\n  ,\n}\n\n/* Resize and reposition favicons */\n.Tab .fav {\n  margin-left: 15px;\n}\n\n.Tab .fav,\n.Tab .placeholder,\n.Tab .t-box {\n  margin-bottom: -3px;\n}\n\n.TabsPanel .container {\n  box-shadow: none;\n}\n\n.Sidebar::before {\n  --border-margin: 16px;\n  content: '';\n  width: calc(100% - var(--border-margin));\n  margin: 0 calc(var(--border-margin) / 2);\n  border-top: 1px solid var(--chrome-content-separator-color);\n}\n\n\n.Tab .audio {\n  z-index: 21 !important;\n  transform: translateX(5px);\n}\n\n.Tab .close {\n  margin: 0 5px;\n}\n\n.Tab .t-box {\n  width: calc(100% - 50px);\n}\n\n.NavigationBar .panels-box {\n  /*  flex-wrap: nowrap;*/\n}\n\n#root[data-nav-layout=\"horizontal\"] .NavigationBar,\n#root[data-nav-layout=\"hidden\"] .NavigationBar {\n  box-shadow: none;\n}\n\n.NavigationBar .panel-btn {\n  top: 5px;\n  left: 5px;\n  width: calc(var(--nav-btn-width) - 6px);\n  height: calc(var(--nav-btn-height) - 6px);\n  border-radius: 4px;\n  margin-right: 5px;\n  margin-bottom: 10px;\n}\n\n.NavigationBar .panel-btn[data-active=\"true\"] {\n  box-shadow: 0 0 1px rgba(128, 128, 142, 0.9);\n}\n\n\n.Sidebar .panel::before{\n  --border-margin: 16px;\n  content: '';\n  width: calc(100% - var(--border-margin));\n  margin: 0 calc(var(--border-margin) / 2);\n  border-top: 1px solid var(--chrome-content-separator-color);\n}\n.NavigationBar .panels-box {\n  flex-wrap: nowrap;\n}\n\n",
    "groupCSS": false,
    "snapNotify": false,
    "snapExcludePrivate": false,
    "snapInterval": 15,
    "snapIntervalUnit": "min",
    "snapLimit": 512,
    "snapLimitUnit": "kb",
    "hScrollThroughPanels": true,
    "scrollThroughTabs": "none",
    "scrollThroughVisibleTabs": false,
    "scrollThroughTabsSkipDiscarded": false,
    "scrollThroughTabsExceptOverflow": true,
    "scrollThroughTabsCyclic": true,
    "tabDoubleClick": "pin",
    "tabLongLeftClick": "mute",
    "tabLongRightClick": "none",
    "tabsPanelLeftClickAction": "none",
    "tabsPanelDoubleClickAction": "tab",
    "tabsPanelRightClickAction": "menu",
    "tabsPanelMiddleClickAction": "tab",
    "syncName": "GloMBP-SB4",
    "syncSaveSettings": true,
    "syncSaveCtxMenu": true,
    "syncSaveStyles": true,
    "syncAutoApply": false
  },
  "permissions": {
    "allUrls": true,
    "tabHide": true,
    "actualAllUrls": true,
    "actualTabHide": true
  },
  "storage": {
    "size": "2.7 mb",
    "props": {
      "bookmarksMenu": "197 b",
      "bookmarksPanelMenu": "54 b",
      "containers_v4": "4 kb",
      "cssVars": "1.6 kb",
      "expandedBookmarks": "18 b",
      "favAutoCleanTime": "10 b",
      "favUrls": "34.1 kb",
      "favicons": "2.2 mb",
      "lastSnapTime": "13 b",
      "panelIndex": "1 b",
      "panels_v4": "2.4 kb",
      "prevTabsData_v4": "1 kb",
      "profileID": "14 b",
      "settings": "7 kb",
      "sidebarCSS": "4.2 kb",
      "snapshots_v4": "508.4 kb",
      "tabsData_v4": "1 kb",
      "tabsMenu": "428 b",
      "tabsPanelMenu": "184 b"
    }
  },
  "panels": [
    {
      "type": "default",
      "id": "firefox-default",
      "name": 7,
      "icon": "...",
      "cookieStoreId": "firefox-default",
      "lockedTabs": true,
      "lockedPanel": false,
      "skipOnSwitching": true,
      "noEmpty": true,
      "newTabCtx": "firefox-default",
      "dropTabCtx": "firefox-default",
      "moveTabCtx": "firefox-default",
      "moveTabCtxNoChild": true
    },
    {
      "type": "tabs",
      "id": "7ktCmOUSv42E",
      "name": 14,
      "icon": "...",
      "color": "...",
      "customIconSrc": "",
      "customIcon": "",
      "lockedTabs": true,
      "lockedPanel": true,
      "skipOnSwitching": false,
      "noEmpty": true,
      "newTabCtx": "firefox-container-2",
      "dropTabCtx": "firefox-container-2",
      "moveTabCtx": "firefox-container-2",
      "moveTabCtxNoChild": true,
      "urlRulesActive": false,
      "urlRules": 147
    },
    {
      "type": "tabs",
      "id": "bs_nrlgUH541",
      "name": 5,
      "icon": "...",
      "color": "...",
      "customIconSrc": "",
      "customIcon": "",
      "lockedTabs": false,
      "lockedPanel": false,
      "skipOnSwitching": false,
      "noEmpty": false,
      "newTabCtx": "firefox-container-28",
      "dropTabCtx": "firefox-container-28",
      "moveTabCtx": "firefox-container-28",
      "moveTabCtxNoChild": true,
      "urlRulesActive": false,
      "urlRules": ""
    },
    {
      "type": "tabs",
      "id": "cWPd7FpO7a51",
      "name": 5,
      "icon": "...",
      "color": "...",
      "customIconSrc": "",
      "customIcon": "",
      "lockedTabs": false,
      "lockedPanel": true,
      "skipOnSwitching": false,
      "noEmpty": false,
      "newTabCtx": "firefox-container-29",
      "dropTabCtx": "firefox-container-29",
      "moveTabCtx": "firefox-container-29",
      "moveTabCtxNoChild": true,
      "urlRulesActive": false,
      "urlRules": ""
    },
    {
      "type": "tabs",
      "id": "nn3y6UAR4pii",
      "name": 5,
      "icon": "...",
      "color": "...",
      "customIconSrc": "",
      "customIcon": "",
      "lockedTabs": false,
      "lockedPanel": true,
      "skipOnSwitching": false,
      "noEmpty": false,
      "newTabCtx": "firefox-container-30",
      "dropTabCtx": "firefox-container-30",
      "moveTabCtx": "firefox-container-30",
      "moveTabCtxNoChild": true,
      "urlRulesActive": false,
      "urlRules": ""
    },
    {
      "type": "tabs",
      "id": "HOBDBteFj4Nh",
      "name": 12,
      "icon": "...",
      "color": "...",
      "customIconSrc": "",
      "customIcon": "",
      "lockedTabs": false,
      "lockedPanel": true,
      "skipOnSwitching": false,
      "noEmpty": false,
      "newTabCtx": "firefox-container-18",
      "dropTabCtx": "firefox-container-18",
      "moveTabCtx": "firefox-container-18",
      "moveTabCtxNoChild": false,
      "urlRulesActive": false,
      "urlRules": 58
    },
    {
      "type": "bookmarks",
      "id": "bookmarks",
      "cookieStoreId": "bookmarks",
      "name": 9,
      "icon": "...",
      "bookmarks": true,
      "lockedPanel": false,
      "skipOnSwitching": false
    }
  ],
  "containers": [
    {
      "id": "firefox-container-1",
      "name": 8,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": "direct",
      "includeHostsActive": true,
      "includeHosts": 330,
      "excludeHostsActive": true,
      "excludeHosts": 83,
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-2",
      "name": 4,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": true,
      "includeHosts": 147,
      "excludeHostsActive": true,
      "excludeHosts": 83,
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-13",
      "name": 21,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-14",
      "name": 18,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-27",
      "name": 12,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-18",
      "name": 11,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": "direct",
      "includeHostsActive": true,
      "includeHosts": 50,
      "excludeHostsActive": true,
      "excludeHosts": 50,
      "userAgentActive": true,
      "userAgent": ""
    },
    {
      "id": "firefox-container-19",
      "name": 8,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-28",
      "name": 5,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-29",
      "name": 5,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-30",
      "name": 5,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-31",
      "name": 7,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-32",
      "name": 8,
      "icon": "...",
      "color": "...",
      "proxified": false,
      "proxy": null,
      "includeHostsActive": false,
      "includeHosts": "",
      "excludeHostsActive": false,
      "excludeHosts": "",
      "userAgentActive": false,
      "userAgent": ""
    }
  ],
  "cssVars": {},
  "sidebarCSSLen": 4132,
  "windows": [
    {
      "state": "maximized",
      "incognito": false,
      "tabsCount": 11
    }
  ],
  "tabsMenu": [
    [
      {
        "name": "Undo-menu"
      },
      "undoRmTab",
      "mute",
      "reload",
      "bookmark"
    ],
    "separator-1",
    [
      {
        "name": "Move to"
      },
      "moveToNewWin",
      "moveToWin",
      "moveToPanel"
    ],
    [
      {
        "name": "Containers"
      },
      "reopenInCtr"
    ],
    "separator-2",
    "duplicate",
    "copyUrls",
    "pin",
    "separator-3",
    [
      {
        "name": "Tabs functions"
      },
      "group",
      "flatten",
      "discard",
      "clearCookies"
    ],
    "separator-4",
    "close",
    "closeDescendants",
    "closeTabsAbove",
    "closeTabsBelow",
    "reopenInNewWin",
    "reopenInWin",
    "closeOtherTabs"
  ],
  "bookmarksMenu": [
    [
      {
        "name": "Open in"
      },
      "openInNewWin",
      "openInNewPrivWin"
    ],
    [
      {
        "name": "Containers"
      },
      "openInCtr"
    ],
    "separator-5",
    "createBookmark",
    "createFolder",
    "createSeparator",
    "separator-7",
    "copyUrls",
    "edit",
    "delete"
  ],
  "bookmarks": {
    "bookmarksCount": 821,
    "foldersCount": 57,
    "separatorsCount": 2,
    "maxDepth": 6
  }
}
</code></details>

<details><summary><b>Logs</b></summary><code>

no messages appear
</code></details>

OK so basically my company started switching globalprotect vpn to use system default browser.
only i have sideberry keep my work related url's in a work container.

what i can see happens, is on browser first launch, it opens the page in the default container.

it then tries to connect to my companies url's for okta sign on, however it auto switches to the work container, and in that switch, the cookie/session info is lost, and so the vpn is unable to connect.

i have tried to set up excludes for panels for file:// stuff, and other bits and bobs, but no joy, doesn't work.

only unblocker is for me to disable side berry when switching to vpn.

not ideal as it means i might just keep side berry disabled perma if i goto keep doing this.

The whole point of side berry is so it can conveniently switch my containers for me, but if it gets in the way i need to lose it :(

i basically need a way to tell it NEVER TO SWITCH away from default unless i move a container to a panel manually, and then always do that.

or alternatively, how about me having regex matches for default container?

if i could make my default container the work one, that might work?

**Comment by mbnuqw** [comment]
If I correctly understood, you can try to use a default container as a "work" container and setup "Exclude URLs" in "personal" container with work URLs.
